From b9bf555e37b304194c602e4eeb806fad45ad77ba Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sun, 30 Jan 2022 07:40:19 -0500
Subject: [PATCH 21/27] Fix parsing audio hal 7.0 audio policies

Google changed separator from natural "," to weird " "
We broke its support in "FIH devices: Fix "Earpiece" audio output"

Change-Id: I458b8b6a6498dd9cf748e00843ff65c561579902
---
 .../audiopolicy/common/managerdefinitions/src/Serializer.cpp    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/audiopolicy/common/managerdefinitions/src/Serializer.cpp b/services/audiopolicy/common/managerdefinitions/src/Serializer.cpp
index 9addf7dbdc..9d3d9fd169 100644
--- a/services/audiopolicy/common/managerdefinitions/src/Serializer.cpp
+++ b/services/audiopolicy/common/managerdefinitions/src/Serializer.cpp
@@ -428,7 +428,7 @@ PolicySerializer::deserialize<AudioProfileTraits>(
     std::string samplingRates = getXmlAttribute(cur, Attributes::samplingRates);
     std::string format = getXmlAttribute(cur, Attributes::format);
     std::string channels = getXmlAttribute(cur, Attributes::channelMasks);
-    ChannelTraits::Collection channelsMask = channelMasksFromString(channels, ",");
+    ChannelTraits::Collection channelsMask = channelMasksFromString(channels, mChannelMasksSeparator.c_str());
 
     //Some Foxconn devices have wrong earpiece channel mask, leading to no channel mask
     if(channelsMask.size() == 1 && *channelsMask.begin() == AUDIO_CHANNEL_IN_MONO && isOutput) {
-- 
2.34.1

